apps/design/backend/src/app.ts

```diff
@@ -893,15 +950,26 @@ export function buildApp(context: AppContext): Application {
   const api = buildApi(context);
   app.use('/api', grout.buildRouter(api, express));
 
-  /* istanbul ignore next - @preserve */
-  if (authEnabled()) {
-    app.use('*', requiresAuth());
-  }
-
   app.use(
     express.static(context.workspace.assetDirectoryPath, { index: false })
   );
 
+  /* istanbul ignore next - @preserve */
+  if (authEnabled()) {
+    app.use('*', (req, res, next) => {
+      if (
+        // Allow unauthenticated access to fetch the html for /report and any api endpoints it calls.
+        req.originalUrl.startsWith('/report') ||
+        METHODS_THAT_DO_NOT_REQUIRE_OAUTH_INTEGRATION.map(
+          (m) => `/api/${m}`
+        ).includes(req.originalUrl)
```

Comment:
I'm pretty sure that because we call `app.use('*', requiresAuth());` after mounting the grout API, this exception shouldn't be necessary. I think it only applies to other routes defined afterwards (i.e. the static routes, the index route).